cylc/flow/network/server.py

```diff
@@ -367,41 +402,24 @@ def graphql(
         Returns:
             object: Execution result, or a list with errors.
         """
-        try:
-            executed: 'ExecutionResult' = schema.execute(
-                request_string,
-                variable_values=variables,
-                context_value={
-                    'resolvers': self.resolvers,
-                    'meta': meta or {},
-                },
-                backend=CylcGraphQLBackend(),
-                middleware=list(instantiate_middleware(self.middleware)),
-                executor=AsyncioExecutor(),
-                validate=True,  # validate schema (dev only? default is True)
-                return_promise=False,
-            )
-        except Exception as exc:
-            return 'ERROR: GraphQL execution error \n%s' % exc
+        executed: 'ExecutionResult' = schema.execute(
+            request_string,
+            variable_values=variables,
+            context_value={
+                'resolvers': self.resolvers,
+                'meta': meta or {},
+            },
+            backend=CylcGraphQLBackend(),
+            middleware=list(instantiate_middleware(self.middleware)),
+            executor=AsyncioExecutor(),
+            validate=True,  # validate schema (dev only? default is True)
+            return_promise=False,
+        )
         if executed.errors:
-            errors: List[Any] = []
             for error in executed.errors:
-                LOG.error(error)
-                if hasattr(error, '__traceback__'):
-                    import traceback
-                    formatted_tb = traceback.format_exception(
-                        type(error), error, error.__traceback__
-                    )
-                    LOG.error("".join(formatted_tb))
-                    errors.append({
-                        'error': {
-                            'message': str(error),
-                            'traceback': formatted_tb
-                        }
-                    })
-                    continue
-                errors.append(getattr(error, 'message', None))
-            return errors
+                LOG.warning(error)
+            if not executed.data:
+                raise Exception(executed.errors[0])
```

Comment:
Where does this error get caught?

Reply:
By graphql

Reply:
Right, GraphQL then swallows this error and injects it into its response?

You've probably already tested this, if not make sure it's nailed down (we don't want the potential for a bad request taking down the server).

Reply:
I think this one is sufficient
https://github.com/cylc/cylc-flow/blob/f751949dfed0005e069ec4052592409fb80d2fb2/tests/integration/network/test_server.py#L138-L146